<?xml version="1.0" encoding="utf-8"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro">  
  <xacro:macro name="Plastic_Storage_Crate" params="handle parent position orientation">

  <joint name="Plastic_Storage_Crate_${handle}_joint" type="fixed">
    <origin xyz="${position}" rpy="${orientation}"/>
    <parent link="${parent}"/>
    <child link="Plastic_Storage_Crate_${handle}"/>
  </joint>

    <link
      name="Plastic_Storage_Crate_${handle}">
      <inertial>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <mass
          value="0" />
        <inertia
          ixx="0"
          ixy="0"
          ixz="0"
          iyy="0"
          iyz="0"
          izz="0" />
      </inertial>
      <visual>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <geometry>
          <mesh
            filename="package://ICRA2023_Quadruped_Competition/meshes/Plastic_Storage_Crate.stl"/>
        </geometry>
        <material
          name="">
          <color
            rgba="0.22 0.76 0.94 1" />
        </material>
      </visual>
      <collision>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <geometry>
          <mesh
            filename="package://ICRA2023_Quadruped_Competition/meshes/Plastic_Storage_Crate.stl" />
        </geometry>
      </collision>
    </link>
  
  </xacro:macro>
</robot>